src/day15.scala

```diff
@@ -0,0 +1,69 @@
+// using scala 3.1.0
+
+package day15
+
+import scala.util.Using
+import scala.io.Source
+import scala.collection.mutable
+
+type Coord = (Int, Int)
+class GameMap(cells: Seq[Seq[Int]]):
+  val maxRow = cells.length - 1
+  val maxCol = cells.head.length - 1
+
+  def neighboursOf(c: Coord): List[Coord] = c match
+    case (row, col) =>
+      val lb = mutable.ListBuffer.empty[Coord]
+      if row < maxRow then lb.append((row+1, col))
+      if row > 0      then lb.append((row-1, col))
+      if col < maxCol then lb.append((row, col+1))
+      if col > 0      then lb.append((row, col-1))
+      lb.toList
+
+  def costOf(c: Coord): Int = c match
+    case (row, col) => cells(row)(col)
```

Comment:
In this case, pattern match saves us one line of code, and the program is one statement instead of two. Now that we don't have braces it's less relevant but it could've been a bigger deal on Scala 2 where with `match` you wouldn't need to use braces.